model_zoo/gpt/run_pretrain.py

```diff
@@ -12,32 +13,68 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import argparse
 import math
 import os
 import random
 import time
+from dataclasses import dataclass, field
+from typing import Optional
 
+import lr
 import numpy as np
 import paddle
-from visualdl import LogWriter
-from paddlenlp.transformers import GPTModel, GPTForPretraining, GPTPretrainingCriterion
-from paddlenlp.transformers import GPTTokenizer, GPTChineseTokenizer
-from paddlenlp.utils.log import logger
-from paddlenlp.utils import profiler
-from paddlenlp.ops import Topology
+from args import parse_config_file
+from dataset import GPTDataset, get_train_valid_test_split_
 
-from dataset import create_pretrained_dataset
-from args import parse_args
-import lr
-from paddle.distributed import fleet
+from paddlenlp.ops import Topology
+from paddlenlp.trainer import (
+    PdArgumentParser,
+    Trainer,
+    TrainingArguments,
+    get_last_checkpoint,
+    speed_metrics,
+)
+from paddlenlp.transformers import (
+    GPTChineseTokenizer,
+    GPTForPretraining,
+    GPTPretrainingCriterion,
+    GPTTokenizer,
+)
+from paddlenlp.utils.batch_sampler import DistributedBatchSampler
+from paddlenlp.utils.downloader import get_path_from_url_with_filelock
+from paddlenlp.utils.log import logger
 
 MODEL_CLASSES = {
     "gpt": (GPTForPretraining, GPTTokenizer),
     "gpt-cn": (GPTForPretraining, GPTChineseTokenizer),
 }
 
 
+@dataclass
+class ModelArguments(TrainingArguments):
+    model_name_or_path: str = field(default="gpt2-en", metadata={"help": ""})
+    max_seq_len: int = field(default=128, metadata={"help": "max sequence length"})
+    input_dir: str = field(default="", metadata={"help": "input idr of dataset"})
+    to_static: bool = field(default=False, metadata={"help": "whether use static pretraining mode."})
+    min_lr: float = field(default=1e-5, metadata={"help": "The initial min learning rate for Adam."})
+    split: str = field(default="949,50,1", metadata={"help": "Train/valid/test data split."})
+    lr_decay_style: str = field(default="none", metadata={"help": "style of learning rate decay"})
+    use_amp: bool = field(default=False, metadata={"help": "whether use amp mode."})
+
+    scale_loss: float = field(
+        default=32768, metadata={"help": "The value of scale_loss for fp16. This is only used for AMP training."}
+    )
```

Comment:
很多参数在 trainer 里面已经有了。
```suggestion
    model_name_or_path: str = field(default="gpt2-en", metadata={"help": ""})
    max_seq_len: int = field(default=128, metadata={"help": "max sequence length"})
    input_dir: str = field(default="", metadata={"help": "input idr of dataset"})
    to_static: bool = field(default=False, metadata={"help": "whether use static pretraining mode."})
    min_lr: float = field(default=1e-5, metadata={"help": "The initial min learning rate for Adam."})
    split: str = field(default="949,50,1", metadata={"help": "Train/valid/test data split."})
```

Reply:
比如剩下的几个参数
ModelArguments
```python
    model_name_or_path: str = field(default="gpt2-en", metadata={"help": ""})
    max_seq_len: int = field(default=128, metadata={"help": "max sequence length"})
    to_static: bool = field(default=False, metadata={"help": "whether use static pretraining mode."})
```

TrainingArguments
```python
    min_lr: float = field(default=1e-5, metadata={"help": "The initial min learning rate for Adam."})
```
DataArguments
```python
    input_dir: str = field(default="", metadata={"help": "input idr of dataset"})
    split: str = field(default="949,50,1", metadata={"help": "Train/valid/test data split."})
```

Reply:
你的建议是将这个拆分成不同的 dataclass 来处理吧。

Reply:
我在最新的commit 当中拆分成不同args 来处理了。